### PR TITLE
add status bar pinned action aria label

### LINKED ISSUE #227257: [Accessibility] Status bar footer needs aria-label info
-     VS Code Version: 1.88.0
-     OS Version: MacOS Sonoma (v14.6.1)

Issue: The status bar <footer> element in VSCode provided with role="status" and tabindex="0" lacks name information. 

Impact: Screen reader users will have difficulty determining the purpose and state of these controls.

Recommendation: Provide descriptive name as "Status bar" (aria-label = "Status bar") or similar for focusable element with role="status".

[Compliant Code Example]
```
<footer aria-label="Status bar" class="part statusbar status-border-top" id="workbench.parts.statusbar" role="status" aria-live="off" data-keybinding-context="8" tabindex="0" (...)>
(...)
</footer>
```
Screenshot of issue:
Screenshot 2024-09-02 at 10 00 56 PM

### LINKED ISSUE #227537: [Accessibility] Update aria-label for the pin "Add to status bar" under "Editor Languages status" {{}) control at the bottom of status bar.
- VS Code Version: 1.88.0
- OS Version: MacOS Sonoma (v14.6.1)

**Issue**: There are identical accessible names under "{} Javascript" inside the status bar at the bottom. Instead update the aria-label = "Add JavaScript version to Status Bar" accessible name.

**[Recommendation]**
Ensure headings and labels are descriptive and unique. This includes the accessible names of buttons. The accessible name of a button can be set with internal text, an aria-label attribute, or an `aria-labelledby` attribute. Good accessible names are both concise and descriptive. Avoid including the word "button" in the accessible name, as this information is already supplied by the button's role.

**Actual behaviour:** The accessibility user finds it difficult to determine between the two controls with same accessible name.
**Expected behaviour:** Need unique and descriptive accessible name for the pins under status bar's "{} Javascript" tab.

**[Compliant Code Example]**
```
<a class="action-label codicon codicon-pin" role="button" aria-label="Add JavaScript version to Status Bar" tabindex="0"></a>
```

Screenshot of the issue:
Screenshot 2024-09-04 at 12 22 18 AM

### PR DESCRIPTION
**Pull request by meganrogge**
fix https://github.com/microsoft/vscode/issues/227537

cc @bpasero

### CODE REVIEW

File: src/vs/workbench/browser/parts/statusbar/statusbarPart.ts [hunk 1]
```diff
@@ -678,6 +678,7 @@ export class AuxiliaryStatusbarPart extends StatusbarPart implements IAuxiliaryS
 		@IContextKeyService contextKeyService: IContextKeyService,
 	) {
 		const id = AuxiliaryStatusbarPart.COUNTER++;
+		container.ariaLabel = 'Status bar';
```
Comment (connor4312) [inline src/vs/workbench/browser/parts/statusbar/statusbarPart.ts:681]: This and the below usage should probably be localized

File: src/vs/workbench/browser/parts/statusbar/statusbarPart.ts [hunk 2]
```diff
@@ -712,6 +712,7 @@ export class StatusbarService extends MultiWindowParts<StatusbarPart> implements
 		statusbarPartContainer.style.position = 'relative';
 		statusbarPartContainer.setAttribute('aria-live', 'off');
 		statusbarPartContainer.setAttribute('tabindex', '0');
+		statusbarPartContainer.setAttribute('aria-label', localize('statusBar', 'Status Bar'));
```
Comment (meganrogge) [inline src/vs/workbench/browser/parts/statusbar/statusbarPart.ts:715]: somehow this is still not getting applied to the element 🤔

Comment (meganrogge) [reply src/vs/workbench/browser/parts/statusbar/statusbarPart.ts:715]: works when I make the change inline in dev tools though

**Review by connor4312 — DISMISSED** [review]

File: src/vs/workbench/browser/parts/statusbar/statusbarPart.ts (hunk 2 above)
Comment (meganrogge) [reply src/vs/workbench/browser/parts/statusbar/statusbarPart.ts]: removed that and removing the issue it fixes as I can't figure it out

**Review by bpasero — APPROVED** [review]

**Pull request merged**